### PR TITLE
support reading config in csi cinder driver

### PR DESCRIPTION
**Pull request by edisonxiang**
support reading config in csi cinder driver

### CODE REVIEW

**Comment by edisonxiang** [comment]
/cc @rootfs @chakri-nelluri
hey rootfs and chakri, welcome to review, thank you.

File: app/cinderplugin/main.go [hunk 1]
```diff
@@ -44,6 +45,9 @@ func main() {
 	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", "", "CSI endpoint")
 	cmd.MarkPersistentFlagRequired("endpoint")
 
+	cmd.PersistentFlags().StringVar(&config, "config", "", "CSI drive config")
```
Comment (rootfs) [inline app/cinderplugin/main.go]: cloud config, consistent with k8s options

File: pkg/cinder/openstack/openstack.go [hunk 2]
```diff
@@ -38,15 +40,93 @@ type OpenStack struct {
...
+		TenantName:       cfg.Global.TenantName,
+		DomainID:         cfg.Global.DomainId,
+		DomainName:       cfg.Global.DomainName,
+
+		// Persistent service, so we need to be able to renew tokens.
+		AllowReauth: true,
+	}
+}
+
+func getConfigFromFile(configFilePath string) (gophercloud.AuthOptions, string, error) {
```
Comment (rootfs) [inline pkg/cinder/openstack/openstack.go]: why is `region` so special in returns?

Comment (rootfs) [inline pkg/cinder/openstack/openstack.go]: have a unit test for this too

Comment (chakri-nelluri) [reply pkg/cinder/openstack/openstack.go]: +1

Comment (edisonxiang) [reply pkg/cinder/openstack/openstack.go]: Thanks. I will.

File: app/cinderplugin/main.go (hunk 1 above)
Comment (edisonxiang) [reply app/cinderplugin/main.go]: Thanks. I will

File: pkg/cinder/openstack/openstack.go (hunk 2 above)
Comment (edisonxiang) [reply pkg/cinder/openstack/openstack.go]: ```region``` is not included in ```gophercloud.AuthOptions```, region is included in ```gophercloud.EndpointOpts```, I guess it is better to return ```gophercloud.AuthOptions``` and ```gophercloud.EndpointOpts```, Thanks your suggestion @rootfs

**Comment by edisonxiang** [comment]
@chakri-nelluri Thanks your perfect suggestion, I will consider it in another pr:)

**Comment by edisonxiang** [comment]
Hi @rootfs @chakri-nelluri, done, PTAL. thanks.

**Comment by rootfs** [comment]
would love to have a followup PR with instructions and screenshot.

**Pull request merged**